### PR TITLE
TGP-1409 & TGP-1665 /assessment radio to bullets and cya changes

### PR DESCRIPTION
**Pull request by Ewan-James-Donovan**
https://jira.tools.tax.service.gov.uk/browse/TGP-1409
https://jira.tools.tax.service.gov.uk/browse/TGP-1665

### CODE REVIEW

File: app/controllers/AssessmentController.scala [hunk 1]
```diff
@@ -98,21 +94,19 @@ class AssessmentController @Inject() (
...
             .fold(
               formWithErrors => {
-                val radioOptions = AssessmentAnswer.radioOptions(exemptions)
-                val viewModel    = AssessmentViewModel(
-                  commodityCode = categorisationInfo.commodityCode,
-                  numberOfThisAssessment = index + 1,
-                  numberOfAssessments = categorisationInfo.categoryAssessments.size,
-                  radioOptions = radioOptions
-                )
+                val listItems = exemptions.map { exemption =>
```
Comment (liandra-giacomine) [inline app/controllers/AssessmentController.scala]: this function could maybe be part of the exemption class? Not too bothered either way though just saw the code duplication

def itemDescription = code + " - " +  description

exemptions.map(_.itemDescription)

File: app/forms/AssessmentFormProvider.scala [hunk 2]
```diff
@@ -24,14 +24,17 @@ import javax.inject.Inject
...
+    else { "assessment.error.required" }
     Form(
       "value" ->
-        text("assessment.error.required")
+        text(messagesKey)
           .verifying(
-            "assessment.error.required",
-            value => exemptionIds.contains(value) || value == AssessmentAnswer.NoExemption.toString
+            messagesKey,
+            value => value == "true" || value == AssessmentAnswer.NoExemption.toString
```
Comment (liandra-giacomine) [inline app/forms/AssessmentFormProvider.scala]: do we still need this bit

|| value == AssessmentAnswer.NoExemption.toString

File: app/viewmodels/checkAnswers/AssessmentsSummary.scala [hunk 3]
```diff
@@ -17,45 +17,41 @@
...
-    answers.get(AssessmentPage(recordId, indexOfThisAssessment)).flatMap { answer =>
-      val descriptiveText = if (answer == NoExemption) {
-        Some("assessment.exemption.none.checkYourAnswersLabel")
-      } else {
-        assessment.exemptions
-          .find(x => x.id == answer.toString)
-          .map(x => messages("assessment.exemption", x.code, x.description))
-      }
+    answers.get(AssessmentPage(recordId, indexOfThisAssessment)).map { answer =>
+      val listItems = assessment.exemptions
```
Comment (liandra-giacomine) [inline app/viewmodels/checkAnswers/AssessmentsSummary.scala]: been asked in the PRA to not use x as variable names, can we change it to something more descriptive? :D

File: app/controllers/AssessmentController.scala (hunk 1 above)
Comment (Ewan-James-Donovan) [reply app/controllers/AssessmentController.scala]: Yep! Moved to CategoryAssessment.scala as we need it for all exemptions, not just a single one.

File: app/forms/AssessmentFormProvider.scala (hunk 2 above)
Comment (Ewan-James-Donovan) [reply app/forms/AssessmentFormProvider.scala]: Not really! :) Refactored

File: app/viewmodels/checkAnswers/AssessmentsSummary.scala (hunk 3 above)
Comment (Ewan-James-Donovan) [reply app/viewmodels/checkAnswers/AssessmentsSummary.scala]: Sure! Did away with the variable entirely.

**Review by liandra-giacomine — APPROVED** [review]

**Review by vinoocs — APPROVED** [review]

**Pull request merged**